### PR TITLE
Added ACM test for various configurations

### PR DESCRIPTION
**Pull request by lazpavel**
#### Description of changes
- added ACM test for various configurations
- contains changes from [9859](https://github.com/aws-amplify/amplify-cli/pull/9859) [only last commit needs reviewed]

#### Checklist
- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-graphql-auth-transformer/src/__tests__/acm.test.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { IndexTransformer, PrimaryKeyTransformer } from '@aws-amplify/graphql-index-transformer';
+import { ModelTransformer } from '@aws-amplify/graphql-model-transformer';
+import { GraphQLTransform } from '@aws-amplify/graphql-transformer-core';
+import { AuthTransformer } from '..';
+import { ACMTest, acmTests } from './acm-test-database';
```
Comment (cjihrig) [inline packages/amplify-graphql-auth-transformer/src/__tests__/acm.test.ts:5]: Micro-nit: "database" doesn't seem like the right word here. Maybe something like "library" - but feel free to ignore me 😄

**Review by cjihrig — APPROVED** [review]
Second commit LGTM

File: packages/amplify-graphql-relational-transformer/src/graphql-many-to-many-transformer.ts [hunk 2]
```diff
@@ -42,13 +42,13 @@ export class ManyToManyTransformer extends TransformerPluginBase {
...
   private hasOneTransformer: HasOneTransformer;
-  private authTransformer: AuthTransformer;
+  private authTransformer: TransformerAuthProvider;
 
   constructor(
     modelTransformer: ModelTransformer,
     indexTransformer: IndexTransformer,
     hasOneTransformer: HasOneTransformer,
-    authTransformer: AuthTransformer,
+    authTransformer: TransformerAuthProvider,
```
Comment (sundersc) [inline packages/amplify-graphql-relational-transformer/src/graphql-many-to-many-transformer.ts:51]: Can this variable be renamed to `authProvider` for clarity?

File: packages/amplify-graphql-relational-transformer/src/graphql-many-to-many-transformer.ts [hunk 3]
```diff
@@ -42,13 +42,13 @@ export class ManyToManyTransformer extends TransformerPluginBase {
   private modelTransformer: ModelTransformer;
   private indexTransformer: IndexTransformer;
   private hasOneTransformer: HasOneTransformer;
-  private authTransformer: AuthTransformer;
+  private authTransformer: TransformerAuthProvider;
```
Comment (sundersc) [inline packages/amplify-graphql-relational-transformer/src/graphql-many-to-many-transformer.ts:45]: nit: Can we rename the variable to `authProvider`?

**Review by sundersc — APPROVED** [review]
Just a nit on variable naming. LGTM otherwise.

**Pull request merged**